app/actions/views/channel.js

```diff
@@ -155,6 +155,26 @@ export function loadPostsIfNecessary(channelId) {
     };
 }
 
+export function loadPostsIfNecessaryWithRetry(channelId) {
+    return (dispatch, getState) => {
+        const state = getState();
+        const {posts, postsInChannel} = state.entities.posts;
+
+        const postsIds = postsInChannel[channelId];
+
+        // Get the first page of posts if it appears we haven't gotten it yet, like the webapp
+        if (!postsIds || postsIds.length < ViewTypes.POST_VISIBILITY_CHUNK_SIZE) {
+            getPostsWithRetry(channelId)(dispatch, getState);
```

Comment:
Is there any point in returning the promises from `getPostsWithRetry` or `getPostsSinceWithRetry`, or do they complete automatically because they're optimistic?